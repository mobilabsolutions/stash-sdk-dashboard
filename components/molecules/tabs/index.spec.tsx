import { testRender, deepRender } from '../../../test_utils'
import TabPanel from './index'
import { useState } from 'react'
import { fireEvent } from 'react-testing-library'

const Container = () => {
  const [active, setActive] = useState(0)
  const tabs = [
    {
      title: 'Tab 1',
      render: () => <div>Tab 1 Content</div>
    },
    {
      title: 'Tab 2',
      render: () => <div>Tab 2 Content</div>
    },
    {
      title: () => <span>Tab 3</span>,
      titleStyle: { borderTop: '1px solid red' },
      render: () => <div>Tab 3 content</div>
    }
  ]
  return <TabPanel active={active} setActive={setActive} tabs={tabs} />
}

it('Tab Should Render only active tab', () => testRender(Container, {}))

it('Tab Should change tab on click event', () => {
  const { getByTestId } = deepRender(Container, {})
  expect(getByTestId('tab-content').firstChild).toMatchSnapshot(
    'Active tab content should be "Tab 1"'
  )
  fireEvent(
    getByTestId('tab-2'),
    new MouseEvent('click', {
      bubbles: true,
      cancelable: true
    })
  )
  expect(getByTestId('tab-content').firstChild).toMatchSnapshot(
    'Active tab content should be now "Tab 3"'
  )
})
