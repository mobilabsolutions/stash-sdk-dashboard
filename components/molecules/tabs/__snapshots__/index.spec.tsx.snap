// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab Should Render only active tab 1`] = `
<div
  className="tabs__Container-sc-12d1d6a-2 eyFFwE"
>
  <div
    className="tabs__TabSection-sc-12d1d6a-1 iSVFeh"
  >
    <span
      className="tabs__Tab-sc-12d1d6a-0 fdeaHf"
      data-testid="tab-0"
      onClick={[Function]}
    >
      Tab 1
    </span>
    <span
      className="tabs__Tab-sc-12d1d6a-0 fYbykR"
      data-testid="tab-1"
      onClick={[Function]}
    >
      Tab 2
    </span>
    <span
      className="tabs__Tab-sc-12d1d6a-0 fYbykR"
      data-testid="tab-2"
      onClick={[Function]}
    >
      Tab 3
    </span>
  </div>
  <div
    data-testid="tab-content"
  >
    <div>
      Tab 1 Content
    </div>
  </div>
</div>
`;

exports[`Tab Should change tab on click event: Active tab content should be "Tab 1" 1`] = `
<div>
  Tab 1 Content
</div>
`;

exports[`Tab Should change tab on click event: Active tab content should be now "Tab 3" 1`] = `
<div>
  Tab 3 content
</div>
`;
